rust-arroyo/src/processing/dlq.rs

```diff
@@ -403,6 +404,8 @@ impl<TPayload> BufferedMessages<TPayload> {
             return;
         }
 
+        let cloned = message.clone();
```

Comment:
let's not clone the entire message. you only need the index (not for example, the entire payload), so re-bind that to a new variable here and use it later.

```
let partition_index = message.partition.index;
```